index.js

```diff
@@ -87,11 +87,19 @@ if (program.opts().block2) {
 }
 
 if (typeof program.opts().coapOption !== 'undefined' && program.opts().coapOption.length > 0) {
+  const options = {}
+  // calling req.setOption() multiple times for the same key will overwrite previous values,
+  // therefore group options by name/key and add them all at once
   program.opts().coapOption.forEach(function (singleOption) {
     const i = singleOption.indexOf(coapOptionSeperator)
-    const kvPair = [singleOption.slice(0, i), singleOption.slice(i + 1)]
-    const optionValueBuffer = kvPair[1].startsWith('0x') ? Buffer.from(kvPair[1].substring(2), 'hex') : Buffer.from(kvPair[1])
-    req.setOption(kvPair[0], optionValueBuffer)
+    const key = singleOption.slice(0, i)
+    const value = singleOption.slice(i + 1)
+    const valueBuffer = value.startsWith('0x') ? Buffer.from(value.substring(2), 'hex') : Buffer.from(value)
+    if (!Object.prototype.hasOwnProperty.call(options, key)) options[key] = []
+    options[key].push(valueBuffer)
+  })
```

Comment:
Maybe there could be an additional check added that ensures that the option is actually repeatable? (See https://datatracker.ietf.org/doc/html/rfc7252#section-5.4.5).

Reply:
Good point, I don't think we should though. If we start validating options we should validate all of them properly, including allowed values, etc. I don't think we want or need that in this package.
Just for reference, the packages `node-coap` and `coap-packet` also don't do any of that.